packages/frontend-shared/src/composables/useCohorts.ts

```diff
@@ -0,0 +1,83 @@
+import { useMutation, gql } from '@urql/vue'
+import { UseCohorts_DetermineCohortDocument } from '../generated/graphql'
+import { ref } from 'vue'
+
+gql`
+mutation UseCohorts_DetermineCohort ($name: String!, $cohorts: [String!]!) {
+  determineCohort(cohortConfig: { name: $name, cohorts: $cohorts } ) {
+    __typename
+    name
+    cohort
+  }
+}`
+
+/**
+ * An option to use for a given cohort selection.
+ */
+export type CohortOption = {
+  /** The individual cohort identifier.  Example: 'A' or 'B' */
+  cohort: string
+
+  /** The value to be used by the calling code for the given cohort. The algorithm for selecting the cohort does not care about this value, but it will return the entire CohortOption that is selected so that this value can be used by the calling code. */
+  value: any
+}
+
+/**
+ * The configuration of the cohort to be used to select a cohort.
+ */
+export type CohortConfig = {
+  /** The name of the feature the cohort will be calculated for.  This will be used as a key in the cache file for storing the selected option. */
+  name: string
+
+  /** Array of options to pick from when selecting the cohort */
+  options: CohortOption[]
+
+  /** Optional array of weights to use for selecting the cohort.  If not supplied, an even weighting algorithm will be used. */
+  weights?: number[]
+}
+
+/**
+ * Composable that encapsulates the logic for choosing a cohort from a list of configured options.
+ *
+ * @remarks
+ * The logic for this composable will first check the cache file to determine if a cohort has already been saved for the given cohort `name`. If found, that cohort will be returned.  If not found or the option found does not match an existing option, a weighted algorithm will be used to pick from the list of CohortOptions. The picked value will be stored in the cache and returned.
+ *
+ * @returns object with getCohort function for returning the cohort
+ */
+export const useCohorts = () => {
```

Comment:
After the refactoring, I feel like the better pattern here would have been to use `pinia`, where the `cohortOptionSelected` would have been stored as `state` and the `getCohort` would be an `action`. It seems strange that a `composable` is returning a function and not reactive state.

Reply:
Can I `store` make a GraphQL mutation?  That is what I was trying to encapsulate in `useCohorts`?

Reply:
You should be able to use `useMutation` inside an action, see https://pinia.vuejs.org/core-concepts/actions.html#accessing-other-stores-actions where they utilize composables inside pinia actions.